src/intervals/interval_rounding.jl

```diff
@@ -0,0 +1,193 @@
+
+## Macros for directed rounding:
+
+# Use the following empty definitions for rounding types other than Float64:
+Base.set_rounding(whatever, rounding_mode) = ()
+Base.get_rounding(whatever) = ()
+
+if VERSION > v"0.4-"
+    Base.Rounding.get_rounding_raw(whatever) = ()
+    Base.Rounding.set_rounding_raw(whatever, rounding_mode) = ()
+end
+
+
+const INTERVAL_ROUNDING = [:narrow]  # or :wide
+# TODO: replace by an enum in 0.4
+
+@doc doc"""`get_interval_rounding()` returns the current interval rounding mode.
+There are two possible rounding modes:
+
+- :narrow  -- changes the floating-point rounding mode to `RoundUp` and `RoundDown`.
+This gives the narrowest possible interval.
+
+- :wide -- Leaves the floating-point rounding mode in `RoundNearest` and uses
+`prevfloat` and `nextfloat` to achieve directed rounding. This creates an interval of width 2`eps`.
+""" ->
+
+get_interval_rounding() = INTERVAL_ROUNDING[end]
+
+set_interval_rounding(mode) = INTERVAL_ROUNDING[end] = mode  # a symbol
+
+
+
+set_interval_rounding(:narrow)
+
+
+macro with_rounding(T, expr, rounding_mode)
+    quote
+        with_rounding($T, $rounding_mode) do
+            $expr
+        end
+    end
+end
+
+
+@doc doc"""The `@round` macro creates a rounded interval according to the current interval rounding mode.
+It is the main function used to create intervals in the library (e.g. when adding two intervals, etc.).
+It uses the interval rounding mode (see get_interval_rounding())""" ->
+macro round(T, expr1, expr2)
+    #@show "round", expr1, expr2
+    quote
+        mode = get_interval_rounding()
+
+        if mode == :wide  #works with any rounding mode set, but the result will depend on the rounding mode
+            # we assume RoundNearest
+            Interval(prevfloat($expr1), nextfloat($expr2))
+
+        else # mode == :narrow
+            Interval(@with_rounding($T, $expr1, RoundDown), @with_rounding($T, $expr2, RoundUp))
+        end
+
+    end
+end
+
+@doc doc"""`@thin_round` possibly saves one operation compared to `@round`.""" ->
+macro thin_round(T, expr)
+    quote
+        mode = get_interval_rounding()
+
+        if mode == :wide  #works with any rounding mode set, but the result will depend on the rounding mode
+            # we assume RoundNearest
+            temp = $expr  # evaluate the expression
+            Interval(prevfloat(temp), nextfloat(temp))
+
+        else # mode == :narrow
+            Interval(@with_rounding($T, $expr, RoundDown), @with_rounding($T, $expr, RoundUp))
+
+        end
+    end
+end
+
+
+@doc doc"""`make_interval` is used by `@interval` to create intervals from individual elements of different types"""->
+make_interval(::Type{BigFloat}, x::String)    =  @thin_round(BigFloat, @compat parse(BigFloat,x))
+
+make_interval(::Type{BigFloat}, x::String)    =  @thin_round(BigFloat, BigFloat(x))
```

Comment:
I think this line was supposed to be replaced by the definition `make_interval(::Type{BigFloat}, x::String)` two lines above; it triggers a deprecation warning.